src/ecs/resources/input/axis.rs

```diff
@@ -0,0 +1,11 @@
+use super::Button;
+
+/// Represents an axis made up of digital inputs, like W and S or A and D.
+/// Two of these could be analogous to a DPAD.
+#[derive(Serialize, Deserialize)]
+pub struct Axis {
+    /// Positive button, when pressed down axis value will return 1 if `neg` is not pressed down.
```

Comment:
@torkleyy This particular Axis struct?  Not really.  Although I do see the merit in introducing a more abstract Axis API so I'll consider that.